pkg/common/helper_test.go

```diff
@@ -0,0 +1,167 @@
+	var err error
+
+	// create a temp directory
+	suite.tempDir, err = ioutil.TempDir("", "isfile-test")
+	suite.Require().NoError(err)
+}
+
+func (suite *IsFileTestSuite) TestIsFileFuncPositive() {
+	//create temp file
+	tempFile, err := ioutil.TempFile(suite.tempDir, "temp_file")
+	suite.Require().NoError(err)
+	suite.Assert().True(IsFile(tempFile.Name()))
+	defer os.Remove(tempFile.Name())
+}
+
+func (suite *IsFileTestSuite) TestIsFileFuncFileIsNotExist() {
+	//create temp file
+	tempFile, err := ioutil.TempFile(suite.tempDir, "temp_file")
+	suite.Require().NoError(err)
+	os.Remove(tempFile.Name())
+	suite.Assert().False(IsFile(tempFile.Name()))
+}
+
+func (suite *IsFileTestSuite) TestIsFileFuncFileIsADirectory() {
+	suite.Assert().False(IsFile(suite.tempDir))
+}
+
+func (suite *IsFileTestSuite) TearDown() {
+	defer os.RemoveAll(suite.tempDir)
+}
+
+func (suite *IsDirTestSuite) TestIsDirFuncPositive() {
+	var err error
+
+	// create a temp directory
+	suite.tempDir, err = ioutil.TempDir("", "isdir-test")
+	suite.Require().NoError(err)
+	suite.Assert().True(IsDir(suite.tempDir))
+}
+
+func (suite *IsDirTestSuite) TestIsDirFuncNegative() {
+	//create temp file
+	tempFile, err := ioutil.TempFile(suite.tempDir, "temp_file")
+	suite.Require().NoError(err)
+	suite.Assert().False(IsDir(tempFile.Name()))
+	defer os.Remove(tempFile.Name())
+}
+
+func (suite *IsDirTestSuite) TearDown() {
+	defer os.RemoveAll(suite.tempDir)
+}
+
+func (suite *FileExistTestSuite) SetupTest()  {
+	var err error
+
+	// create a temp directory
+	suite.tempDir, err = ioutil.TempDir("", "file_exist-test")
+	suite.Require().NoError(err)
+}
+
+func (suite *FileExistTestSuite) TestFileExistsFuncPositive() {
+	//create temp file
+	tempFile, err := ioutil.TempFile(suite.tempDir, "temp_file")
+	suite.Require().NoError(err)
+	suite.Assert().True(IsFile(tempFile.Name()))
+	defer os.Remove(tempFile.Name())
+}
+
+func (suite *FileExistTestSuite) TestFileNotExist() {
+	//create temp file
+	tempFile, err := ioutil.TempFile(suite.tempDir, "temp_file")
+	suite.Require().NoError(err)
+	os.Remove(tempFile.Name())
+	suite.Assert().False(IsFile(tempFile.Name()))
+}
+
+func (suite *FileExistTestSuite) TestFileExistFuncFileIsNotAFile() {
+	suite.Assert().False(IsFile(suite.tempDir))
+}
+
+func (suite *FileExistTestSuite) TearDown() {
+	defer os.RemoveAll(suite.tempDir)
+}
+
+func (suite *StringSliceToIntSliceTestSuite) TestStringSliceToIntSlicePositive() {
+	stringSlice := []string{"1","2","5","6","23"}
+	expectedSlice := []int{1,2,5,6,23}
+	actualSlice, err := StringSliceToIntSlice(stringSlice);
+	suite.Assert().Equal(expectedSlice,actualSlice)
+	suite.Assert().Equal(nil,err)
+}
+
+func (suite *StringSliceToIntSliceTestSuite) TestStringSliceToIntSliceNegativeData() {
+	stringSlice := []string{"1","2","5","6","23", "someBadData"}
+	_, err := StringSliceToIntSlice(stringSlice);
+	suite.Assert().True(err!=nil)
+}
+
+func (suite *RetryUntilSuccessfulTestSuite) TestRetryUntilSuccessfulPositive() {
```

Comment:
This needs to have much more elaborate testing...
Test # of calls until successful
Time passed between retries
etc